server/services/taxonomyService.ts

```diff
@@ -72,16 +78,72 @@ class TaxonomyService extends BaseService<TaxonomyServiceEvents> {
       this.taxonomy.statusCounts[status] = 0;
     });
 
+    this.taxonomy.locationCounts = {'': 0};
+    this.taxonomy.locationSizes = {};
+    this.taxonomy.locationTree = [];
     this.taxonomy.statusCounts[''] = 0;
     this.taxonomy.tagCounts = {'': 0, untagged: 0};
     this.taxonomy.tagSizes = {};
     this.taxonomy.trackerCounts = {'': 0};
     this.taxonomy.trackerSizes = {};
   };
 
+  buildLocationTree = () => {
+    const locations = Object.keys(this.taxonomy.locationCounts);
+    const sortedLocations = locations.slice().sort((a, b) => {
+      if (a === '') {
+        return -1;
+      }
+      if (b === '') {
+        return 1;
+      }
+
+      // Order slashes before similar paths with different symbols, eg. /files/PC/ should come before /files/PC-98/ for treeing
+      return a.replace(/[^\\/\w]/g, '~').localeCompare(b.replace(/[^\\/\w]/g, '~'));
```

Comment:
```suggestion
      const aa = a.replace(/[^\\/\w]/g, '~');
      const bb = b.replace(/[^\\/\w]/g, '~');
      if (aa < bb) {
        return -1;
      }
      return 1;
```

Reply:
should fix this bug

Reply:
Ah ha, that would do it. Are you running a case-insensitive OS in some manner that's programmatically detectable? Windows is obviously.. Looks like mac is too? Linux might be in a rare case of custom ext4. 

I'm thinking it might be easiest to just add `i` to the regex flags if on a case-insensitive OS... or it looks like even better is passing options to localeCompare.

```suggestion
      const insensitiveOS = ['Darwin', 'Windows_NT'].includes(os.type());
      const compareOptions = {ignorePunctuation: true};
      if (insensitiveOS) {
        compareOptions['sensitivity'] = 'accent';
      }
      // I'm not sure what case I had for replacing everything but word characters and slashes with '~' -- that also drops non-latin characters, which isn't great.
      return a.localeCompare(b, undefined, compareOptions);
```

Reply:
Obviously a handful of test cases for this is in order.

Reply:
won't work

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#sensitivity

Reply:
Why won't it work? Mac and Windows are insensitive, so for insensitive OS, it uses 'accent', which is case insensitive.

Reply:
With 'accent', this array is still not sorted as expected.

Reply:
In my understanding the root case is array being sorted case-insensitive, so we don't need to distinguish os, just need to sort them case-sensitive

Reply:
and also, I'm running flood on a case-sensitive OS, so...

Reply:
I just test this locally, it doesn't work.